### PR TITLE
Issue 6389: LTS - Report percentage used space.

### LINKED ISSUE #6389: LTS - Enable enforcing soft quota on LTS usage and safe mode.
**Is your feature request related to a problem? Please describe.**
Currently there is no detection of LTS full and no reporting of any metrics related to storage usage.

**Describe the solution you'd like**
_New behavior_
1. There is an optional user defined soft limit on total LTS usage by a deployment. 
2. If underlying storage provider can provide quick LTS usage stats, then it should be checked on regular basis.
3. If current usage is above max safe LTS usage size then LTS enters a safe mode.
4. In safe mode write and concat operations on **non-system** segments throw StorageFullException. 
5. When the usage drops below limit the safe mode should be exited.

_New member for ChunkedStorage_
   ` CompletableFuture<Long> getUsedSpace();`

_New config properties_

- ` storage.safe.size.check.enable` - whether to enable this feature or not
-  ` storage.safe.size.bytes.max` - soft limit on max size . This limit should leave enough space on LTS for writing system generated data . Suggested size is 90% of actual absolute max.
- ` storage.safe.size.check.frequency.seconds` - how often to do this check.
 
_New metrics_
segmentstore.storage.used_bytes

**Describe alternatives you've considered**
We need a very simple solution to begin with. Any computationally intensive options should be rejected.

**Additional context**

### PR DESCRIPTION
**Pull request by sachin-j-joshi**
Signed-off-by: Sachin Joshi <sachin.joshi@emc.com>

**Change log description**  
Issue 6389: LTS - Report percentage used space.

**Purpose of the change**  
Fixes #6389 

**What the code does**  
Report percentage value for space used.

**How to verify it**  
All tests should pass.

### CODE REVIEW

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java [hunk 1]
```diff
@@ -68,9 +68,7 @@
 import static io.pravega.segmentstore.storage.chunklayer.ChunkStorageMetrics.SLTS_CREATE_LATENCY;
 import static io.pravega.segmentstore.storage.chunklayer.ChunkStorageMetrics.SLTS_DELETE_COUNT;
 import static io.pravega.segmentstore.storage.chunklayer.ChunkStorageMetrics.SLTS_DELETE_LATENCY;
-import static io.pravega.shared.MetricsNames.SLTS_STORAGE_USED_BYTES;
-import static io.pravega.shared.MetricsNames.STORAGE_METADATA_NUM_CHUNKS;
-import static io.pravega.shared.MetricsNames.STORAGE_METADATA_SIZE;
+import static io.pravega.shared.MetricsNames.*;
```
Comment (andreipaduroiu) [inline segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java:71]: I thought we disallow star imports. Does checkstyle like this?

File: segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageMockTests.java [hunk 2]
```diff
@@ -435,8 +434,17 @@ public void testUpdateStorageStatsException() throws Exception {
...
 
     @Test
-    public void testReport() {
-        val config = ChunkedSegmentStorageConfig.DEFAULT_CONFIG;
+    public void testReport() throws Exception {
+
+        MetricsConfig metricsConfig = MetricsConfig.builder().with(MetricsConfig.ENABLE_STATISTICS, true).build();
+        MetricsProvider.initialize(metricsConfig);
+        StatsProvider statsProvider = MetricsProvider.getMetricsProvider();
+        statsProvider.startWithoutExporting();
```
Comment (andreipaduroiu) [inline segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageMockTests.java:442]: Does this need any cleanup?

Comment (sachin-j-joshi) [reply segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorageMockTests.java:451]: fixed.

File: segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java (hunk 1 above)
Comment (sachin-j-joshi) [reply segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkedSegmentStorage.java]: Apparently checkstyle didn't complain. Looks like rule doesn't apply for static imports.
But fixed it anyway.

**Review by andreipaduroiu — APPROVED** [review]

**Pull request merged**